workflow-templates/python-linters.yaml

```
@@ -0,0 +1,80 @@
+name: Run Python linters
+
+on: [push, pull_request]
+
+jobs:
+  python-files:
+    runs-on: ubuntu-latest
+    outputs:
+      filelist: ${{ steps.python-files.outputs.filelist }}
+    steps:
+
+    - id: python-files
+      run:
+        echo "::set-output name=filelist::$(find . -type f -exec awk ' /^#!.*python/{print FILENAME} {nextfile}' {} +)"
+
+  pylint:
+    runs-on: ubuntu-latest
+    needs: [python-files]
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.6
+
+    - uses: actions/cache@v2
+      with:
+        path: ~/pip-cache
+        key: pip-3.6-${{ github.sha }}
+        # allow cache hits from previous runs of the current branch,
+        # parent branch, then upstream branches, in that order
+        restore-keys: |
+          pip-3.6-
+    - name: Install Requirements
+      run: |
+        python -m pip install --upgrade pip
+        pip --cache-dir ~/pip-cache install pylint htcondor flask jinja2 gunicorn
+    - name: Run Pylint
+      env:
+        PYTHONPATH: $PYTHONPATH:$PWD/src/htcondorce
+        PYTHON_FILES: ${{ needs.python-files.outputs.filelist }}
+      # HACK: in newer versions of HTCondor (at least 8.9.11+),
+      # classad/htcondor modules are placed in python packages where
+      # straight 'import htcondor' works but this causes problems for
+      # pylint unless you use 'from htcondor import htcondor'. Older
+      # versions of HTCondor don't work with the latter so instead of
+      # adding a version requirement to HTCondor-CE, we'll just ignore
+      # these errors
+      run: |
+        pylint --errors-only --disable no-member $PYTHON_FILES
+  flake8:
+    runs-on: ubuntu-latest
+    needs: [python-files]
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.6
+
+    - uses: actions/cache@v2
+      with:
+        path: ~/pip-cache
+        key: pip-3.6-${{ github.sha }}
+        # allow cache hits from previous runs of the current branch,
+        # parent branch, then upstream branches, in that order
+        restore-keys: |
+          pip-3.6-
+    - name: Install Requirements
+      run: |
+        python -m pip install --upgrade pip
+        pip --cache-dir ~/pip-cache install flake8 htcondor flask jinja2 gunicorn
+    - name: Run flake8
+      env:
+        PYTHONPATH: $PYTHONPATH:$PWD/src/htcondorce
+        PYTHON_FILES: ${{ needs.python-files.outputs.filelist }}
+      run: |
+        flake8 --select F $PYTHON_FILES
```

```suggestion
        flake8 --select F $PYTHON_FILES

```

It's a common convention to ensure that files have a newline at the end